src/qibocal/fitting/classifier/qubit_fit.py

```diff
@@ -0,0 +1,138 @@
+from dataclasses import dataclass
+from math import atan2
+
+import numpy as np
+import numpy.typing as npt
+
+from .utils import identity
+
+
+def constructor(_hyperparams):
+    r"""Return the model class.
+
+    Args:
+        _hyperparams: Model hyperparameters.
+    """
+    return QubitFit()
+
+
+def hyperopt(_x_train, _y_train, _path):
+    r"""Perform an hyperparameter optimization and return the hyperparameters.
+
+    Args:
+        x_train: Training inputs.
+        y_train: Training outputs.
+        path (path): Model save path.
+
+    Returns:
+        Dictionary with model's hyperparameters.
+    """
+    return {}
+
+
+normalize = identity
+
+
+@dataclass
+class QubitFit:
+    r"""This class deploys a qubit state classifier.
+
+    Args:
+        iq_mean0 (np.ndarray): Center of the ground state cloud in the I-Q plane.
+        iq_mean1 (np.ndarray): Center of the excited state cloud in the I-Q plane.
+        threshold (float): Classifier's threshold.
+        angle (float): Rotational angle.
+    """  # TODO: add references
+
+    iq_mean0: np.ndarray = np.array([0.0, 0.0])
+    iq_mean1: np.ndarray = np.array([0.0, 0.0])
+    threshold: float = 0.0
+    angle: float = 0.0
+    fidelity: float = None
+    assignment_fidelity: float = None
+
+    def fit(self, iq_coordinates, states: list):
+        r"""Evaluate the model's parameters given the
+        `iq_coordinates` and their relative ``states`
+        (reference: <https://arxiv.org/abs/1004.4323>).
+        """
+        nshots = len(iq_coordinates)
+        iq_state1 = iq_coordinates[(states == 1)]
+        iq_state0 = iq_coordinates[(states == 0)]
+        self.iq_mean0 = np.mean(iq_state0, axis=0)
+        self.iq_mean1 = np.mean(iq_state1, axis=0)
+
+        # translate
+        iq_coordinates_translated = self.translate(iq_coordinates)
+        iq_state1_trans = self.translate(self.iq_mean1)
+        self.angle = -1 * atan2(iq_state1_trans[1], iq_state1_trans[0])
+
+        # rotate
+        iq_coord_rot = self.rotate(iq_coordinates_translated)
+
+        x_values_state0 = iq_coord_rot[(states == 0)][:, 0]
+        x_values_state1 = iq_coord_rot[(states == 1)][:, 0]
+
+        x_values_state0 = np.sort(x_values_state0)
+        x_values_state1 = np.sort(x_values_state1)
```

Comment:
```suggestion
        x_values_state0 = np.sort(iq_coord_rot[(states == 0)][:, 0])
        x_values_state1 = np.sort(iq_coord_rot[(states == 1)][:, 0])
```

Just to avoid shadowing, but it is really the same.